HPCC-13515 Updates to readme to provide a more general overview, and links to details.

boneill42 opened the pull request.

boneill42 committed:
Updates to detail dependencies required for Linux build.
(used base Ubuntu 14 to test)

boneill42 committed:
Update to the documentation for "make install" (since it requires nodejs)

boneill42 committed:
A proper README.md, moving old README.md -> BUILD_ME.md

This README contains links into the online documentation, and to the wiki for thorough build instructions.

boneill42 committed:
Removed double header lines.

richardkchapman commented on README.md:
```diff
@@ -1,102 +1,63 @@
+
+## [Roxie](http://hpccsystems.com/FAQ/what-roxie)
+
+Roxie (the Query Cluster) provides separate high-performance online query processing and data warehouse capabilities.  Roxie (Rapid Online XML Inquiry Engine) is the data delivery engine used in HPCC to serve data quickly and can support many thousands of requests per node per second. 
```
Roxie stands for Richard's Online XML Inquiry Engine (despite what some of our literature may claim)

richardkchapman commented on README.md:
```diff
@@ -1,102 +1,63 @@
+* Non-procedural and dataflow oriented
+* Modular, reusable, extensible syntax
+* Combines data representation and algorithm implementation
+* Easily extend using C++ libraries
```
Worth mentioning that you can embed Python, JavaScript, etc

boneill42 replied:
Agreed, will update.

mckellyln commented on README.md:
```diff
@@ -1,102 +1,63 @@
+
+Thor (the Data Refinery Cluster) is responsible for consuming vast amounts of data, transforming, linking and indexing that data. It functions as a distributed file system with parallel processing power spread across the nodes. A cluster can scale from a single node to thousands of nodes.
+
+* Single-threaded
```
Thor is multi-threaded

richardkchapman replied:
I think the distinction Brian is trying to make is that thor runs one query at a time while Roxie runs multiple. Internally, both use multiple threads (and multiple processes) to get the job done.

mckellyln replied:
ok, I concur.  But perhaps threaded is not the best word to use here (but I cannot think of anything better at the moment).

richardkchapman replied:
Multiple concurrent queries / Single job runs at a time

boneill42 commented:
Sorry -- I can't take any credit here...

I actually just copied content from this page:
http://hpccsystems.com/Why-HPCC/How-it-works

richardkchapman commented:
@boneill42 Looks like this is waiting for a couple of minor updates from you before I merge it.

AttilaVamos commented:
Automated Smoketest Build: success

boneill42 committed:
HPCC-13515 : Addressing comments and merging latest master

Merge branch 'master' of github.com:hpcc-systems/HPCC-Platform into better_readme

Conflicts:
	README.md

boneill42 commented:
@richardkchapman Comments should be address with the latest commit.

richardkchapman commented:
You should not merge latest master into your branch - rather, you should rebase your branch so that it is relative to latest master. That way your branch will only have the changes you have made rather than a second copy of all the other changes (which would get very confusing).

richardkchapman commented:
Cleaned up and merged manually to master
